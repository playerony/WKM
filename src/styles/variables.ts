export interface Breakpoint {
  largeScreen: number
  mediumScreen: number
  smallScreen: number
  mobileScreen: number
}

export interface Padding {
  big: number
  small: number
  medium: number
}

export type Font = 'normal' | 'bold'

export type Color = 'red' | 'blue' | 'gray' | 'black' | 'cream' | 'white' | 'orange' | 'purple'

export interface Variable {
  baseFontSize: number
  color: {
    [key in Color]: string
  }
  font: {
    [key in Font]: string
  }
  padding: Padding
  breakpoint: Breakpoint
}

const variables: Variable = {
  baseFontSize: 16,
  color: {
    red: '#E81F25',
    blue: '#1890ff',
    gray: '#AAAAAA',
    black: '#16161a',
    cream: '#FBF7F3',
    white: '#FFFFFF',
    orange: '#E4572E',
    purple: '#6B5B95'
  },
  font: {
    normal: `'Montserrat', sans-serif`,
    bold: "'HelveticaNeueBold', Helvetica, Arial, sans-serif"
  },
  breakpoint: {
    largeScreen: 1919,
    mediumScreen: 1367,
    smallScreen: 1023,
    mobileScreen: 750
  },
  padding: {
    small: 20,
    medium: 40,
    big: 60
  }
}

export default variables
