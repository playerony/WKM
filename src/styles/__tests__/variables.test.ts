import variables from '../variables'

import snapshotTest from '@tests/snapshotTest'

describe('variables file', () => {
  snapshotTest(variables)
})
