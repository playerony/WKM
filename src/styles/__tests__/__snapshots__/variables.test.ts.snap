// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variables file snapshot test should match 1`] = `
Object {
  "baseFontSize": 16,
  "breakpoint": Object {
    "largeScreen": 1919,
    "mediumScreen": 1367,
    "mobileScreen": 750,
    "smallScreen": 1023,
  },
  "color": Object {
    "black": "#000000",
    "blue": "#1890ff",
    "cream": "#FBF7F3",
    "gray": "#AAAAAA",
  },
  "font": Object {
    "normal": "'Montserrat', sans-serif",
  },
  "padding": Object {
    "big": 60,
    "medium": 40,
    "small": 20,
  },
}
`
